### PR TITLE
[RISCV] Remove incomplete PRE_DEC/POST_DEC code for XTHeadMemIdx.

### PR DESCRIPTION
**Pull request by topperc**
As far as I can tell if getIndexedAddressParts received an ISD::SUB, the constant would be negated. So `IsInc` should be set to true since the SUB was effectively converted to ADD.  This means we should never use PRE_DEC/POST_DEC.

No tests are affected because DAGCombine aggressively turns SUB with constant into ADD so no lit test has a SUB reach getIndexedAddressParts.

### CODE REVIEW

**Review by wangpc-pp — COMMENTED** [review]
I don't see any error and it seems that AArch64 is the same.
@zixuan-wu Please help double check this, thanks!

File: llvm/lib/Target/RISCV/RISCVISelLowering.cpp [hunk 1]
```diff
@@ -19288,7 +19287,8 @@ bool RISCVTargetLowering::getIndexedAddressParts(SDNode *Op, SDValue &Base,
     if (!isLegalIndexedOffset)
       return false;
 
-    IsInc = (Op->getOpcode() == ISD::ADD);
+    // Constant for SUB was negated earlier.
+    IsInc = true;
```
[the remark this answers is not in the document]
Comment (ptomsich) [reply llvm/lib/Target/RISCV/RISCVISelLowering.cpp:19291]: Good catch!

**Review by ptomsich — APPROVED** [review]
Reviewed and LGTM.
I didn't rerun tests, as that was already done by Craig.

**Review by asb — APPROVED** [review]
LGTM.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Pull request merged**